### PR TITLE
feat: capi initial commit and baseline results

### PR DESCRIPTION
**Pull request by utxrsh**
This PR
- adds initial pretraining plus eval setup(same as MAE) for [CAPI](https://arxiv.org/abs/2502.08769)
- ViT-B model with num_frames set to 16 and used as channels dim (same as setting temporal patch size to 16) on fMRI flat maps, trained for 100 epochs with batch size set to 32 (625k iters, same as MAE baseline)
  - basic evals: image

### CODE REVIEW

File: src/flat_capi/engine_pretrain.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        predict_indices: Absolute indices of tokens to predict.
+        visible_indices: Absolute indices of visible tokens.
+        target: Teacher assignments used as soft labels.
+        temp: Student softmax temperature.
+
+    Returns:
+        Mean loss tensor (double).
+    """
+    core = model.module if hasattr(model, "module") else model
+    amp_dtype = images.dtype if images.dtype in (torch.float16, torch.bfloat16) else torch.float32
```
Comment (clane9) [inline src/flat_capi/engine_pretrain.py]: I think inputs are not supposed to be cast (see [here](https://docs.pytorch.org/docs/stable/amp.html#torch.autocast)). So you will be disabling amp by default.
